### PR TITLE
Extensions: Zoninator - Add routing & tabs navigation

### PR DESCRIPTION
**Pull request by ice9js**
This PR adds navigation and sets up routing for Zoninator's settings:

screen shot 2017-07-12 at 18 11 53

These settings are now also accessible through "Edit plugin settings" on the plugin details page:

screen shot 2017-07-12 at 18 12 45

# Testing

- Head over to *Zone Manager* (Zoninator) plugin details.
- Click **Edit plugin settings**.
- On the settings page, ensure the tabs as well as the url are changing correctly when you click through them.

### CODE REVIEW

File: client/extensions/zoninator/components/navigation/index.jsx [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+Navigation.defaultProps = {
+	activeTab: '',
+};
+
+const connectComponent = connect( ( state ) => {
+	const siteId = getSelectedSiteId( state );
+
+	return {
+		siteSlug: getSiteSlug( state, siteId ),
```
Comment (ockham) [inline client/extensions/zoninator/components/navigation/index.jsx]: Turns out there's a `getSelectedSiteSlug` shorthand selector that we can use.

File: client/extensions/zoninator/components/navigation/index.jsx [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+import SectionNav from 'components/section-nav';
+import SectionNavTabs from 'components/section-nav/tabs';
+import SectionNavTabItem from 'components/section-nav/item';
+import SectionNavSegmented from 'components/section-nav/segmented';
+import Button from 'components/button';
+import sectionsModule from 'sections';
+import { getSiteSlug } from 'state/sites/selectors';
+import { getSelectedSiteId } from 'state/ui/selectors';
+
+const getZones = () => {
```
Comment (ockham) [inline client/extensions/zoninator/components/navigation/index.jsx]: Does this need to be a function?

File: client/extensions/zoninator/components/navigation/index.jsx [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+					{ translate( 'New Zone' ) }
+				</Button>
+			</SectionNavSegmented>
+		</SectionNav>
+	);
+};
+
+Navigation.propTypes = {
+	activeTab: PropTypes.string,
+	site: PropTypes.object,
```
Comment (ockham) [inline client/extensions/zoninator/components/navigation/index.jsx]: Is there a `site` prop at all?

File: client/extensions/zoninator/components/settings/index.jsx [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+		<Main className={ 'zoninator__main' }>
+			<DocumentHead title={ translate( 'WP Zone Manager' ) } />
+			<Navigation activeTab={ tab } />
+			{ children }
+		</Main>
+	);
+};
+
+Settings.propTypes = {
+	site: PropTypes.object,
```
Comment (ockham) [inline client/extensions/zoninator/components/settings/index.jsx]: Is there? 😄

**Comment by donnapep** [comment]
Looks like there's a failing test we should try to fix.

**Comment by ockham** [comment]
> Looks like there's a failing test we should try to fix.

`className` lint error :slightly_frowning_face: I'd opt for ignoring...

File: client/extensions/zoninator/components/navigation/index.jsx (hunk 2 above)
Comment (ice9js) [inline client/extensions/zoninator/components/navigation/index.jsx]: It's a placeholder for now and will eventually be replaced by a Redux selector. I used a function to remind of the dynamic nature of this list.

File: client/extensions/zoninator/components/settings/index.jsx [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+import Main from 'components/main';
+import Navigation from '../navigation';
+
+const Settings = ( {
+	children,
+	tab,
+	translate
+} ) => {
+	return (
+		<Main className={ 'zoninator__main' }>
```
Comment (donnapep) [inline client/extensions/zoninator/components/settings/index.jsx]: No need for the curlies; this is just a plain old string. But this is likely the cause of the linting issue and why CCI is failing. You can store the class name in a variable and that should silence the linter. It's a bit of a hack, but better than having CircleCI fail. See https://github.com/Automattic/wp-calypso/blob/master/client/extensions/wp-job-manager/components/settings/index.jsx#L43.

**Comment by donnapep** [comment]
> className lint error 🙁 I'd opt for ignoring...

This doesn't happen in WPJM though, so I think it can be fixed. See https://github.com/Automattic/wp-calypso/pull/16146#pullrequestreview-49731796.

File: client/extensions/zoninator/components/navigation/index.jsx (hunk 1 above)
Comment (donnapep) [inline client/extensions/zoninator/components/navigation/index.jsx]: Oh dang, I didn't use that either. 😛

**Comment by ice9js** [comment]
Looks like I might have forgotten to stage all of my changes after fixing all linter issues 😞 .
Should be good now 😄

**Pull request merged**